### PR TITLE
Refactor `str` paths to `Path`

### PR DESCRIPTION
**Pull request by bbyalcinkaya**
Refactored the `scenario` and `kasmer` modules to use `Path` instead of `str` for variables representing file and directory paths.

### CODE REVIEW

File: kmultiversx/src/kmultiversx/kasmer.py [hunk 1]
```diff
@@ -60,25 +59,8 @@
...
-    test_wasm_path = glob.glob(test_dir + '/output/*.wasm')
-    # TODO this loads the first wasm file in the directory. what if there are multiple wasm files?
-    if test_wasm_path:
-        return test_wasm_path[0]
-    else:
-        raise ValueError(f'WASM file not found: {test_dir}/output/?.wasm')
-
-
-def load_contract_wasms(contract_wasm_paths: Iterable[str]) -> dict[bytes, KInner]:
-    contract_wasm_modules = {bytes(f, 'ascii'): load_wasm(f) for f in contract_wasm_paths}
```
Comment (bbyalcinkaya) [inline kmultiversx/src/kmultiversx/kasmer.py:81]: I'll add these functions to `kasmer-multiversx` in a follow-up PR.

File: kmultiversx/src/kmultiversx/kasmer.py [hunk 2]
```diff
@@ -60,25 +59,8 @@
...
-        raise FileNotFoundError(f'{INPUT_FILE_NAME!r} not found in "{test_dir!r}"') from None
-
-
-def find_test_wasm_path(test_dir: str) -> str:
-    test_wasm_path = glob.glob(test_dir + '/output/*.wasm')
-    # TODO this loads the first wasm file in the directory. what if there are multiple wasm files?
-    if test_wasm_path:
-        return test_wasm_path[0]
-    else:
-        raise ValueError(f'WASM file not found: {test_dir}/output/?.wasm')
```
Comment (bbyalcinkaya) [inline kmultiversx/src/kmultiversx/kasmer.py:77]: I'll add these functions to `kasmer-multiversx` in a follow-up PR.

File: kmultiversx/src/kmultiversx/kasmer.py [hunk 3]
```diff
@@ -167,15 +149,15 @@ def run_config_and_check_empty(
 # Test metadata
 
 
-def get_test_endpoints(test_dir: str) -> Mapping[str, tuple[str, ...]]:
-    abi_paths = glob.glob(test_dir + '/output/*.abi.json')
-    # TODO this loads the first wasm file in the directory. what if there are multiple wasm files?
+def get_test_endpoints(test_dir: Path) -> Mapping[str, tuple[str, ...]]:
+    abi_paths = glob.glob(str(test_dir) + '/output/*.abi.json')
```
Comment (tothtamas28) [inline kmultiversx/src/kmultiversx/kasmer.py:153]: Consider using `Path.glob` instead: https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob

File: kmultiversx/src/kmultiversx/scenario.py [hunk 4]
```diff
@@ -447,15 +448,15 @@ def mandos_to_check_account(address: str, sections: dict, filename: str) -> list
...
+def mandos_to_deploy_tx(tx: dict, filename: Path, output_dir: Path) -> KInner:
     sender = mandos_argument_to_kbytes(tx['from'])
     value = mandos_int_to_kint(get_egld_value(tx))
     arguments = mandos_arguments_to_klist(tx['arguments'])
     gas_limit = mandos_int_to_kint(tx['gasLimit'])
     gas_price = mandos_int_to_kint(tx.get('gasPrice', '0'), default_when_empty=0)
 
     code = get_contract_code(tx['contractCode'], filename)
-    assert isinstance(code, str)
+    assert isinstance(code, Path)
```
Comment (tothtamas28) [inline kmultiversx/src/kmultiversx/scenario.py:459]:
```suggestion
    assert code is not None
```

**Review by tothtamas28 — APPROVED** [review]
Nice refactoring!

**Pull request merged**